File: c#/2750.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        int N = int.Parse(Console.ReadLine());

        int[] Arr = Enumerable.Range(0, N)
            .Select(_ => int.Parse(Console.ReadLine()))
            .ToArray();

        QuickSort(Arr, 0, N - 1);

        Console.WriteLine(string.Join("\n", Arr));
    }

    static void QuickSort(int[] Arr, int Left, int Right)
    {
        if (Left >= Right)
        {
            return;
        }

        int i = Left;
        int j = Right;
        int pivot = Arr[(i + j) / 2];

        while (i <= j)
        {
            while (Arr[i] < pivot)
            {
                i++;
            }
            while (Arr[j] > pivot)
            {
                j--;
            }

            if (i > j)
            {
                break;
            }

            (Arr[i], Arr[j]) = (Arr[j], Arr[i]);
            i++;
            j--;
        }

        QuickSort(Arr, Left, j);
        QuickSort(Arr, i, Right);
    }
}

File: javascript/2750.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

const N = parseInt(input.shift());
const arr = input.map((i) => parseInt(i));

quickSort(arr, 0, N - 1);

console.log(arr.join("\n"));

function quickSort(arr, left, right) {
  if (left >= right) {
    return arr;
  }

  let [i, j] = [left, right];
  const pivot = arr.at((i + j) / 2);

  while (i <= j) {
    while (arr[i] < pivot) {
      i++;
    }
    while (arr[j] > pivot) {
      j--;
    }

    if (i > j) {
      break;
    }

    [arr[i], arr[j]] = [arr[j], arr[i]];
    i++;
    j--;
  }

  quickSort(arr, left, j);
  quickSort(arr, i, right);
}
